r/tests/testthat/test-dataset-csv.R

```diff
@@ -476,3 +476,32 @@ test_that("CSV reading/parsing/convert options can be passed in as lists", {
 
   expect_equal(ds1, ds2)
 })
+
+test_that("open_csv_dataset params passed through to open_dataset", {
+  ds <- open_csv_dataset(csv_dir, partitioning = "part")
+  expect_r6_class(ds$format, "CsvFileFormat")
+  expect_r6_class(ds$filesystem, "LocalFileSystem")
+  expect_identical(names(ds), c(names(df1), "part"))
+  expect_identical(dim(ds), c(20L, 7L))
+
+  # delim
+  ds_tsv <- open_csv_dataset(tsv_dir, delim = "\t", partitioning = "part")
```

Comment:
This kinda begs the question: should there also be `open_tsv_dataset()`, `open_delim_dataset()`, etc.?

Reply:
Yeah, those make sense; added now